lms/djangoapps/courseware/access.py

```diff
@@ -562,6 +565,20 @@ def can_load():
         if not group_access_response:
             return group_access_response
 
+        if is_integrity_signature_enabled():
```

Comment:
I think we should check for integrity signature after both group access and staff access have been checked. We don't want to block staff from seeing content if they have not signed the honor code (is that a correct understanding?)